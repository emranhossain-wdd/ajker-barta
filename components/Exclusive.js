import { formatDistanceToNow } from 'date-fns';
import { useRouter } from 'next/router';
import React, { useState } from 'react'
import styles from '../styles/exclusive.module.css'
const Exclusive = ({ newses }) => {
  const latest = newses?.reverse()
  const router = useRouter()
  const [isLetest, setIsLetest] = useState(true)
  let count = 1
  return (
    <div data-testid='exclusiveId' className='pb-5'>
      <div className="container">
        <div className="grid lg:grid-cols-2 gap-3">
          <div>
            <button onClick={() => setIsLetest(true)} className={isLetest ? 'py-1 slim-border rounded-sm mb-3 px-4 border-blue-400 mr-2 bg-blue-500 text-white' : 'py-1 slim-border rounded-sm mb-3 px-4 bg-white border-blue-400 mr-2'}>Letest</button>
            <button onClick={() => setIsLetest(false)} className={!isLetest ? 'py-1 slim-border rounded-sm mb-3 px-4 border-blue-400 mr-2 bg-blue-500 text-white' : 'py-1 slim-border rounded-sm mb-3 px-4 bg-white border-blue-400 mr-2'}>Trending</button>
            {
              isLetest ? <div className='grid sm:grid-cols-2 gap-3'>
                {
                  latest?.slice(0, 8).map(news => <div onClick={() => router.push(`/${news.category}/${news.subCategory}/${news?._id}`)} className=' relative p-6 shadow-md rounded-r-lg border-r-4 border-red-600 cursor-pointer' key={news._id}>
                    <h1 className='leading-7 relative z-10 text-lg font-medium text-gray-800 hover:text-red-600 transition-colors duration-300 cursor-pointer'>{news?.heading}</h1>
                    <p style={{ transform: "translate(-50%, -50%)", }} className='absolute top-1/2 left-3/4 text-8xl font-bold text-blue-100 '>{count++}</p>
                  </div>)
                }
              </div> : <div className='grid sm:grid-cols-2 gap-3'>
                {
                  latest?.slice(8, 16).map(news => <div onClick={() => router.push(`/news/${news._id}`)} className=' relative p-6 shadow-md rounded-r-lg border-r-4 border-red-600 cursor-pointer' key={news._id}>
                    <h1 className='leading-7 relative z-10 text-lg font-medium text-gray-800 hover:text-red-600 transition-colors duration-300 cursor-pointer'>{news?.heading}</h1>
                    <p style={{ transform: "translate(-50%, -50%)", }} className='absolute top-1/2 left-3/4 text-8xl font-bold text-blue-100 '>{count++}</p>
                  </div>)
                }
              </div>
            }

          </div>
          <div>
            <h1 className={styles.exclusive}>Satireday</h1>
            <div className='grid sm:grid-cols-2 gap-3'>
              {
                latest?.slice(16, 20).map(news => <div className='cursor-pointer imageEffect' onClick={() => router.push(`/${news.category}/${news.subCategory}/${news?._id}`)} key={news?._id}>
                  <div className='overflow-hidden'>
                    <img className='w-full h-32 object-cover' src={news?.images.img1} alt="" />
                  </div>
                  <h1 className='text-lg my-1 font-medium leading-5 hover:text-red-600 transition-colors duration-300 cursor-pointer'>{news?.heading}</h1>
                  <p className='text-sm leading-4'>{news?.description[0]?.slice(0, 100)}</p>
                  <p className="px-2 mt-2 py-1 font-medium text-sm rounded-full bg-gray-100 w-fit text-blue-500">{`${formatDistanceToNow(new Date(news.publishedDate))} ago`}</p>
                </div>)
              }
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

export default Exclusive;