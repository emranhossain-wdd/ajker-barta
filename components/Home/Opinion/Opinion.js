import React from 'react';
import Analysis from './AnaLysis/Analysis';
import Editorial from './Editorial/Editorial';
import Interview from './Interview/Interview';
import OpEd from './OpEd/OpEd';
import WorldMedia from './WorldMedia/WorldMedia';

const Opinion = ({ opinion }) => {
    return (
        <>
            <OpEd opinion={opinion} />
            <div className='container mx-auto lg:pb-5 py-10 border-b'>
                <div className="grid sm:grid-cols-2 md:grid-cols-3 lg:grid-cols-4 gap-2">
                    <Editorial opinion={opinion} />
                    <Interview opinion={opinion} />
                    <Analysis opinion={opinion} />
                    <WorldMedia opinion={opinion} />
                </div>
            </div>
        </>
    );
};

export default Opinion;