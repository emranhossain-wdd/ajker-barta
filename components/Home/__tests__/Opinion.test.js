import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import Opinion from '../Opinion/Opinion'
import { render } from '@testing-library/react'
import { testArr } from '../../__tests__/Exclusive.test'

it('render Opinion component without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(<Opinion />, div)
})

it('testing Opinion with props', () => {
  const { getByTestId } = render(<Opinion testArr={testArr} />)
  expect(getByTestId('opinionId')).toBeTruthy()
})
it('maching snapshot', () => {
  const tree = renderer.create(<Opinion />)
  expect(tree).toMatchSnapshot()
})