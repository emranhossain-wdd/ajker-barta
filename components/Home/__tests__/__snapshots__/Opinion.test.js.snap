// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`maching snapshot 1`] = `
<div
  className="pb-5"
  data-testid="exclusiveId"
>
  <div
    className="container"
  >
    <div
      className="grid lg:grid-cols-2 gap-3"
    >
      <div>
        <button
          className="py-1 slim-border rounded-sm mb-3 px-4 border-blue-400 mr-2 bg-blue-500 text-white"
          onClick={[Function]}
        >
          Letest
        </button>
        <button
          className="py-1 slim-border rounded-sm mb-3 px-4 bg-white border-blue-400 mr-2"
          onClick={[Function]}
        >
          Trending
        </button>
        <div
          className="grid sm:grid-cols-2 gap-3"
        />
      </div>
      <div>
        <h1
          className="exclusive"
        >
          Satireday
        </h1>
        <div
          className="grid sm:grid-cols-2 gap-3"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`maching snapshot 2`] = `
<div
  data-testid="opinionId"
>
  <div
    className="pt-5"
  >
    <div
      className="container"
    >
      <div
        className="flex items-center mb-4"
      >
        <h1
          className="ml-2 text-xl cursor-pointer font-medium text-blue-900"
          onClick={[Function]}
        >
          Opinion 
        </h1>
        <svg
          className="text-red-600 mt-1"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M294.1 256L167 129c-9.4-9.4-9.4-24.6 0-33.9s24.6-9.3 34 0L345 239c9.1 9.1 9.3 23.7.7 33.1L201.1 417c-4.7 4.7-10.9 7-17 7s-12.3-2.3-17-7c-9.4-9.4-9.4-24.6 0-33.9l127-127.1z"
          />
        </svg>
      </div>
      <div
        className="grid grid-cols-12 gap-6"
      >
        <div
          className="lg:col-span-5 col-span-12 border-2 border-gray-200 relative flex items-center ml-8 p-12 cursor-pointer text-lg"
          onClick={[Function]}
          style={
            Object {
              "height": "450px",
            }
          }
        >
          <h1
            className="absolute text-xl top-8 bg-blue-900 px-4 -left-6
 py-2 text-white"
          />
          <p
            className="text-gray-600 mt-4"
          />
          <h2
            className="absolute bottom-5 left-4 text-xl text-white"
          />
        </div>
        <div
          className="lg:col-span-7 col-span-12"
        />
      </div>
    </div>
  </div>
  <div
    className="container mx-auto lg:pb-5 py-10 border-b"
  >
    <div
      className="grid sm:grid-cols-2 md:grid-cols-3 lg:grid-cols-4 gap-2"
    >
      <div>
        <h2
          className="text-xl mb-3 text-red-500 font-serif font-medium cursor-pointer"
          onClick={[Function]}
        >
          <span
            className="text-black"
          >
            Editorial 
          </span>
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ChevronRightIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 6 8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
            />
          </svg>
        </h2>
        <div
          className="border-r pr-2"
        >
          <div
            className="pb-3 cursor-pointer"
            onClick={[Function]}
          >
            <img
              alt=""
              className="rounded-t-md h-52 object-cover w-full"
            />
            <h1
              className="text-xl font-serif my-1 hover:text-blue-500 transition-colors duration-300"
            />
          </div>
          <div
            className="border-t py-2 cursor-pointer"
            onClick={[Function]}
          >
            <h1
              className="text-xl font-serif hover:text-blue-500 transition-colors duration-300"
            >
              <div />
            </h1>
          </div>
          <div
            className="border-t py-2 cursor-pointer"
            onClick={[Function]}
          >
            <h1
              className="text-xl font-serif hover:text-blue-500 transition-colors duration-300"
            >
              <div />
            </h1>
          </div>
        </div>
      </div>
      <div>
        <h2
          className="text-xl mb-3 text-red-500 font-serif font-medium cursor-pointer"
          onClick={[Function]}
        >
          <span
            className="text-black"
          >
            Interview 
          </span>
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ChevronRightIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 6 8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
            />
          </svg>
        </h2>
        <div
          className="border-r pr-2"
        >
          <div
            className="pb-3 cursor-pointer"
            onClick={[Function]}
          >
            <img
              alt=""
              className="rounded-t-md h-52 object-cover w-full"
            />
            <h1
              className="text-xl font-serif my-1 hover:text-blue-500 transition-colors duration-300"
            />
          </div>
          <div
            className="border-t py-2 cursor-pointer"
            onClick={[Function]}
          >
            <h1
              className="text-xl font-serif hover:text-blue-500 transition-colors duration-300"
            >
              <div />
            </h1>
          </div>
          <div
            className="border-t py-2 cursor-pointer"
            onClick={[Function]}
          >
            <h1
              className="text-xl font-serif hover:text-blue-500 transition-colors duration-300"
            >
              <div />
            </h1>
          </div>
        </div>
      </div>
      <div>
        <h2
          className="text-xl mb-3 text-red-500 font-serif font-medium cursor-pointer"
          onClick={[Function]}
        >
          <span
            className="text-black"
          >
            Analysis 
          </span>
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ChevronRightIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 6 8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
            />
          </svg>
        </h2>
        <div
          className="border-r pr-2"
        >
          <div
            className="pb-3 cursor-pointer"
            onClick={[Function]}
          >
            <img
              alt=""
              className="rounded-t-md h-52 object-cover w-full"
            />
            <h1
              className="text-xl font-serif my-1 hover:text-blue-500 transition-colors duration-300"
            />
          </div>
          <div
            className="border-t py-2 cursor-pointer"
            onClick={[Function]}
          >
            <h1
              className="text-xl font-serif hover:text-blue-500 transition-colors duration-300"
            >
              <div />
            </h1>
          </div>
          <div
            className="border-t py-2 cursor-pointer"
            onClick={[Function]}
          >
            <h1
              className="text-xl font-serif hover:text-blue-500 transition-colors duration-300"
            >
              <div />
            </h1>
          </div>
        </div>
      </div>
      <div>
        <h2
          className="text-xl mb-3 text-red-500 font-serif font-medium cursor-pointer"
          onClick={[Function]}
        >
          <span
            className="text-black"
          >
            World Media 
          </span>
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ChevronRightIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 6 8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
            />
          </svg>
        </h2>
        <div>
          <div
            className="pb-3 cursor-pointer"
            onClick={[Function]}
          >
            <img
              alt=""
              className="rounded-t-md h-52 object-cover w-full"
            />
            <h1
              className="text-xl font-serif my-1 hover:text-blue-500 transition-colors duration-300"
            />
          </div>
          <div
            className="border-t py-2 cursor-pointer"
            onClick={[Function]}
          >
            <h1
              className="text-xl font-serif hover:text-blue-500 transition-colors duration-300"
            >
              <div />
            </h1>
          </div>
          <div
            className="border-t py-2 cursor-pointer"
            onClick={[Function]}
          >
            <h1
              className="text-xl font-serif hover:text-blue-500 transition-colors duration-300"
            >
              <div />
            </h1>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
